micro-deps/micro-deps/src/main/java/com/ofg/infrastructure/discovery/MicroserviceConfiguration.java

```diff
@@ -20,12 +22,11 @@
 
     public MicroserviceConfiguration(ServicePath servicePath, List<Dependency> dependencies) {
         this.servicePath = servicePath;
-        this.dependencies = dependencies;
+        this.dependencies = unmodifiableList(dependencies);
     }
 
     public MicroserviceConfiguration(ServicePath servicePath) {
-        this.servicePath = servicePath;
-        this.dependencies = new ArrayList<>();
+        this(servicePath, new ArrayList<Dependency>(0));
```

Comment:
It's not possible to use diamonds in such cases but empty collection is the one that should be used, I don't know how could I missed that